### PR TITLE
Add a submit confirmation modal

### PR DESCRIPTION
**Pull request by mikix**
https://openedx.atlassian.net/browse/DISCO-855

### CODE REVIEW

File: src/components/EditCoursePage/index.jsx [hunk 1]
```diff
@@ -270,6 +291,12 @@ class EditCoursePage extends React.Component {
           <title>{`Course - ${title}`}</title>
         </Helmet>
 
+        <SubmitConfirmModal
+          open={this.state.submitConfirmVisible}
```
Comment (Dillon-Dumesnil) [inline src/components/EditCoursePage/index.jsx]: Please deconstruct this with the rest of the state variables.

File: src/components/SubmitConfirmModal/index.jsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+
+const SubmitConfirmModal = ({
+  onSubmit,
+  ...others
+}) => (
+  <Modal
+    {...others}
+    title="Submit for Review?"
+    body="You will not be able to make edits while the course is in review, which can take up to 48 hours. Confirm your edits are complete."
```
Comment (Dillon-Dumesnil) [inline src/components/SubmitConfirmModal/index.jsx]: Bigger message. More information.

After this point, you can no longer update entitlements (or mode/price or whatever). Maybe something about submitting runs is still allowed?

Maybe talk to Carol about messaging here

Comment (mikix) [reply src/components/SubmitConfirmModal/index.jsx]: Talked to Carol. She isn't so worried about emphasizing that submitting runs is still OK. But likes the idea of warning about mode changes.

But she thought the modal was maybe too late for that and asked it to be like, above the field. I'll look at that change.

Comment (Dillon-Dumesnil) [reply src/components/SubmitConfirmModal/index.jsx]: Okay, I'm cool with whatever. Just thought we should add in a little something.

File: src/components/EditCoursePage/EditCourseForm.jsx [hunk 3]
```diff
@@ -141,7 +141,7 @@ export class BaseEditCourseForm extends React.Component {
     return (
       <div className="edit-course-form">
         <form id={id}>
-          <FieldLabel text="Course" className="mt-4 mb-2 h2" />
+          <FieldLabel text={title} className="mt-4 mb-2 h2" />
```
Comment (mikix) [inline src/components/EditCoursePage/EditCourseForm.jsx]: Unrelated change that Carol asked for.

File: src/components/EditCoursePage/EditCourseForm.jsx [hunk 4]
```diff
@@ -663,6 +663,7 @@ export class BaseEditCourseForm extends React.Component {
                       id="mode.label"
                       text="Enrollment track"
                       helpText={enrollmentHelp}
+                      extraText="(Cannot edit after submission)"
```
Comment (mikix) [inline src/components/EditCoursePage/EditCourseForm.jsx]: Rather than having it in modal, Carol wanted this above field.

**Comment by jlajoie** [comment]
Just a followup for what's going on with testing our RichEditor component with `mount()` is effectively the same as https://github.com/tinymce/tinymce-angular/issues/76. Specifically, the contentWindow does not exist on the HTMLIFrameElement at the time the TinyMCE code references it.
